corehq/apps/user_importer/tasks.py

```diff
@@ -37,10 +42,69 @@ def _update_progress(value, start=0):
         'rows': user_results['rows']
     }
     upload_record = UserUploadRecord.objects.using(DEFAULT_DB_ALIAS).get(pk=upload_record_id)
-    upload_record.task_id = import_users_and_groups.request.id
+    upload_record.task_id = task.request.id
     upload_record.result = results
     upload_record.save()
     DownloadBase.set_progress(task, total, total)
     return {
         'messages': results
     }
+
+
+@task(serializer='pickle', queue='user_import_queue')
+def parallel_import_task(domain, user_specs, group_specs, upload_user, upload_record_id):
+    task = parallel_import_task
+    return import_users_and_groups(domain, user_specs, group_specs, upload_user, upload_record_id, task)
+
+
+@task(serializer='pickle', queue='user_import_queue')
+def parallel_user_import(domain, user_specs, group_specs, upload_user):
+    task = parallel_user_import
+    total = len(user_specs) + len(group_specs)
+    DownloadBase.set_progress(task, 0, total)
+    task_list = []
+    for users in chunked(user_specs, 100):
```

Comment:
I am going to change this to 1000 before merging, but leaving it like this for staging testing.